src/middlewared/middlewared/plugins/ports/ports.py

```diff
@@ -0,0 +1,38 @@
+import copy
+
+from middlewared.service import Service, ValidationErrors
+
+
+class PortService(Service):
+
+    DELEGATES = {}
+    SYSTEM_USED_PORTS = [
+        {'title': 'System', 'ports': [6000], 'namespace': 'system'},
+    ]
+
+    class Config:
+        private = True
+
+    async def register_attachment_delegate(self, delegate):
+        self.DELEGATES[delegate.namespace] = delegate
+
+    async def get_in_use(self):
+        ports = copy.deepcopy(self.SYSTEM_USED_PORTS)
```

Comment:
Why `deepcopy`? Just iterate inside this function and create a new list and then simply `ports += self.SYSTEM_USED_PORTS`